pkg/registry/apis/provisioning/jobs/migrate/worker.go

```diff
@@ -154,17 +156,31 @@ func (w *MigrationWorker) Process(ctx context.Context, repo repository.Repositor
 		}
 	}
 
-	// Load and write all folders
-	progress.SetMessage("exporting folders")
-	err = worker.loadFolders(ctx)
-	if err != nil {
-		return err
-	}
+	if dualwrite.IsReadingLegacyDashboardsAndFolders(ctx, w.storageStatus) {
+		progress.SetMessage("exporting folders")
+		err = worker.loadFolders(ctx)
+		if err != nil {
+			return err
+		}
 
-	progress.SetMessage("exporting resources")
-	err = worker.loadResources(ctx)
-	if err != nil {
-		return err
+		progress.SetMessage("exporting resources")
+		err = worker.loadResources(ctx)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Already using unified storage
```

Comment:
🤔 with unified storage enabled, we may need to *start* with sync disabled so that things do not kick off a sync job while this is running

Reply:
what about using a initialisation state or having a sync status for this? (e.g. pending_initialization)?